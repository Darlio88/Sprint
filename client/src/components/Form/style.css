.form-container{
    display:grid;
    /* place-items: center; */
    /* padding:20%; */
}

.form-container form{
  display:grid;
  gap:2rem;
  width:min(80%, 400px);
  margin-left: auto;
  margin-right:auto;
  color:var(--armygreen);
}

.form-container form > div{
    display:grid;
    gap:0.5rem; 
}

form label{
    font-size: 1.1rem;
    font-weight:500;
}


form input{
    border-radius: 0.25rem;
    padding:0.3rem 0.5rem;
    background-color: var(--lightgreen);
    border:1px solid var(--armygreen);
    color:var(--armygreen);
    outline:transparent;
    
}
input::placeholder{
    color:var(--armygreen);
}