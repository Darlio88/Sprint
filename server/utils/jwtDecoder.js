const jwt = require("jsonwebtoken")

function decodeToken(token, secretKey) {
    try {
      const decoded = jwt.verify(token, secretKey);
      console.log(decoded)
      return decoded;
    } catch (error) {
      // Handle any error that occurs during token verification
      console.error('Token verification failed:', error);
      return null;
    }
  }
  

module.exports=decodeToken;