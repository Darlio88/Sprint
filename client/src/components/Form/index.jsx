import React,{useEffect, useState} from 'react'
import axios from "axios";
import { useNavigate } from 'react-router-dom';
//styles
import "./style.css"
import { Link } from 'react-router-dom'



function Index({type}) {
  const [userName, setUserName]= useState("")
  const [userEmail, setUserEmail]= useState("")
  const [userPassword, setUserPassword]= useState("")
  const navigate = useNavigate()
  async function handleSubmit(e){
    e.preventDefault()
    await axios.post(`http://localhost:4000/users/${type}`,{FullName:userName,Email:userEmail,Password:userPassword}).then(res=>{
      const token = res.data?.token;
      localStorage.setItem("token",token)
      setUserName("")
      setUserEmail("")
      setUserPassword("")
      navigate("/")
    })
  }

  return (
    <section className='form-container'>
       <form onSubmit={handleSubmit}>
       <h3 style={{justifySelf:"center"}} >{type=="signin"?"Log Into Your Account":"Create An account"}</h3>

{  type!=="signin"  ?  <div>
        <label htmlFor="name">Full Name</label>
        <input type="text" value={userName} onChange={(e)=>setUserName(e.target.value)} placeholder='Omoding Daniel' id="name" />
      </div>:null}
      <div>
        <label htmlFor="email">Email</label>
        <input type="email" value={userEmail} onChange={(e)=>setUserEmail(e.target.value)} placeholder='danielomoding173@gmail.com' id="email" />
      </div>
      <div>
        <label htmlFor="password">Password</label>
        <input type="password" value={userPassword} onChange={(e)=>setUserPassword(e.target.value)} placeholder='Password' id="password"/>
      </div>
      <div>
        <button>{type=="signin"?"Login":"Signup"}</button>
        <p>{type=="signin"?"Do not yet":"Already"} have an account? <Link to={type=="signin"?"/signup":"/signin"}>click here</Link></p>
      </div>
    </form>
    </section>
  )
}

export default Index