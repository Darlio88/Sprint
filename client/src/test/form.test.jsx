import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom';
import { render, screen, userEvent } from '../utils/test-utils'

// import Form from '../components/Form'
import Form from "../components/Form"

function toJson(component) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result;
}

test('form contains onsubmit function', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Form type="signin" />
    </BrowserRouter>,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})


describe("form renders well",()=>{
  it("title of form displayed well",()=>{
    render(<BrowserRouter><Form type="signin"/></BrowserRouter>)
    expect(screen.getByText(/Log Into Your Account/i)).toBeInTheDocument()
  })

  
})


