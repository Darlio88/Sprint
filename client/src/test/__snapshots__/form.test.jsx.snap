// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`form contains onsubmit function 1`] = `
<section
  className="form-container"
>
  <form
    onSubmit={[Function]}
  >
    <h3
      style={
        {
          "justifySelf": "center",
        }
      }
    >
      Log Into Your Account
    </h3>
    <div>
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        onChange={[Function]}
        placeholder="danielomoding173@gmail.com"
        type="email"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div>
      <button>
        Login
      </button>
      <p>
        Do not yet
         have an account? 
        <a
          href="/signup"
          onClick={[Function]}
        >
          click here
        </a>
      </p>
    </div>
  </form>
</section>
`;
